admin-dev/themes/new-theme/js/pages/product/edit/combinations-manager.js

```diff
@@ -27,37 +27,101 @@ import ProductMap from '@pages/product/product-map';
 import CombinationsGridRenderer from '@pages/product/edit/combinations-grid-renderer';
 import CombinationsService from '@pages/product/services/combinations-service';
 import DynamicPaginator from '@components/pagination/dynamic-paginator';
+import SubmittableInput from '@components/form/submittable-input';
 
 const {$} = window;
 
 export default class CombinationsManager {
   constructor() {
     this.$productForm = $(ProductMap.productForm);
+    this.$combinationsContainer = $(ProductMap.combinations.combinationsContainer);
+    this.combinationIdInputsSelector = ProductMap.combinations.combinationIdInputsSelector;
     this.initialized = false;
+    this.combinationsService = new CombinationsService(this.getProductId());
+
     this.init();
 
     return {};
   }
 
   init() {
-    const productId = this.getProductId();
     this.paginator = new DynamicPaginator(
       ProductMap.combinations.paginationContainer,
-      new CombinationsService(productId),
+      this.combinationsService,
       new CombinationsGridRenderer(),
     );
-
+    this.initSubmittableInputs();
+    this.onDefaultCombinationSelection();
     // Paginate to first page when tab is shown
     this.$productForm.find(ProductMap.combinations.navigationTab).on('shown.bs.tab', () => this.firstInit());
   }
 
+  initSubmittableInputs() {
+    const combinationToken = this.getCombinationToken();
+
+    new SubmittableInput('.combination-quantity', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[quantity][value]': input.value, 'combination_item[_token]': combinationToken},
+      );
+    });
+
+    new SubmittableInput('.combination-impact-on-price', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[impact_on_price][value]': input.value, 'combination_item[_token]': combinationToken},
+      );
+    });
+
+    new SubmittableInput('.combination-reference', async (input) => {
+      await this.combinationsService.updateListedCombination(
+        this.findCombinationId(input),
+        {'combination_item[reference][value]': input.value, 'combination_item[_token]': combinationToken},
+      );
+    });
+  }
+
+  onDefaultCombinationSelection() {
+    this.$combinationsContainer.on('change', ProductMap.combinations.isDefaultInputsSelector, async (e) => {
+      const checkedInputs = this.$combinationsContainer.find(
+        `${ProductMap.combinations.isDefaultInputsSelector}:checked`,
+      );
+
+      // stop loop when only one input is checked
+      if (checkedInputs.length === 1) {
+        return;
+      }
+
+      await this.combinationsService.updateListedCombination(
```

Comment:
```suggestion
      const selectedDefaultId = this.findCombinationId(e.currentTarget);
      await this.combinationsService.updateListedCombination(
```

You should store this one since it's reused in the loop later